web/vim_lib.js

```diff
@@ -72,6 +75,38 @@ var LibraryVIM = {
 
     dropbox: null,
 
+    // from https://github.com/s-macke/jor1k/blob/master/js/lib/utf8.js
+    // LICENSE: https://github.com/s-macke/jor1k/blob/master/LICENSE.md
+    UnicodeToUTF8Stream: function(key) {
```

Comment:
The problem could be that our string is not null terminated. Now I used the function UTF8ArrayToString instead of Pointer_stringify on a null terminated array and it seems to work fine.

Now I am looking to see if I can make it work with emscripten functions also the other way around.